src/components/Pages/Assets/Tokens/Tokens.tsx

```diff
@@ -0,0 +1,57 @@
+import { BN } from '@polkadot/util';
+import { memo, useEffect } from 'react';
+
+import type { NativeTokenMetadata, TokenBalance, TokenMetadata } from '@helpers/interfaces';
+
+import NativeToken from './NativeToken';
+import Token from './Token';
+
+interface TokensProps {
+  getNativeBalance: () => void;
+  getNativeMetadata: () => void;
+  getTokensBalances: () => void;
+  getTokensMetadata: () => void;
+  nativeBalance: BN | null;
+  nativeMetadata: NativeTokenMetadata | null;
+  tokensBalances: TokenBalance[] | null;
+  tokensMetadata: TokenMetadata[] | null;
+}
+
+const Tokens = ({
+  getNativeBalance,
+  getNativeMetadata,
+  getTokensBalances,
+  getTokensMetadata,
+  nativeBalance,
+  nativeMetadata,
+  tokensBalances,
+  tokensMetadata,
+}: TokensProps) => {
+  useEffect(() => {
+    getNativeBalance();
+    getNativeMetadata();
+    getTokensBalances();
+    getTokensMetadata();
+  }, [getNativeBalance, getNativeMetadata, getTokensBalances, getTokensMetadata]);
```

Comment:
this useEffect scares me :D
Not sure, but I think every method here is called 4 times each